osu.Game/Screens/Select/FooterButton.cs

```diff
@@ -84,6 +84,7 @@ public FooterButton()
         public Action Hovered;
         public Action HoverLost;
         public Key? Hotkey;
+        public bool ExcludePressWithShift;

```

Comment:
I don't like this solution. Can the case where this is used (SelectRandom vs CancelRandom) not be handled inside SongSelect? Maybe rename the method and have something like...
```
var randomButton = new FooterButton(..., Keys.F2, TriggerRandom);

void TriggerRandom(InputState state)
{
    if (state.Keyboard.ShiftPressed)
        CancelRandom();
    else
        SelectRandom();
}
```

I don't like the naming of CancelRandom and SelectRandom either. Perhaps rename them to `SelectPreviousRandom` and `SelectNextRandom` respectively?